src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1263,12 +1262,29 @@ protected override void ApplyDocumentInfoChanged(DocumentId documentId, Document
             // By setting this property, Visual Studio will perform the file rename, which 
             // will cause the workspace's current solution to update and will fire the 
             // necessary workspace changed events.
-            projectItemForDocument.Name = uniqueName;
+            try
```

Comment:
@jasonmalinowski i'm not sure what to do about these.  trying to run this on Roslyn and i def ran into cases where tehse threw.  i think it's all the linked files and shared-code-projects.

Unfortunately, on the feature-side, i'm not sure if there's any way to detect these cases up front and figure out how to avoid them.  Any thoughts from you?

Reply:
@CyrusNajmabadi Do you remember more what the failure was? I could imagine we may have an issue if your refactoring is trying to rename the same file more than once (but it's in different projects), and maybe that's what could be happening. But I have a vague memory of @ryzngard fixing some things there. But that's making a very random guess.

I don't even know if a feature _could_ do filtering for other reasons like shared code projects, since we don't expose that in any way in the workspace layer. And frankly, we'd have to go figure out how to even know those ourselves since that's abstracted away from us too without calling a bunch of legacy VS APIs.

Reply:
tl;dr: let's figure out why it failed, since maybe that's something this layer can deal with. If it can't, then I don't imagine we can push it elsewhere, so maybe just catching the exception would be the right thing to do. It's probably better to apply most of the renames instead of crashing.